signer/src/keys.rs

```diff
@@ -689,4 +764,124 @@ mod tests {
             tweaked_aggregate_key2.0.x_only_public_key().0.serialize();
         assert_eq!(tweaked_aggregate_key1_bytes, tweaked_aggregate_key2_bytes);
     }
+
+    // Helper to create a PrivateKey for testing using a seeded RNG
+    fn sk_from_seed(seed: u64) -> PrivateKey {
+        let mut rng = StdRng::seed_from_u64(seed);
+        PrivateKey::new(&mut rng)
+    }
+
+    // Helper to create a PublicKey for testing using a seeded RNG
+    fn pk_from_seed(seed: u64) -> PublicKey {
+        PublicKey::from_private_key(&sk_from_seed(seed))
+    }
+
+    // Helper to create a bitcoin::BlockHash for testing
+    // Ensures different seeds produce different (though simple) hashes.
+    fn bh_from_seed(seed: u8) -> bitcoin::BlockHash {
+        let mut bytes = [0u8; 32];
+        bytes[0] = seed; // Simple differentiation
+        // For more varied hashes, could use a simple counter or fill more bytes.
+        // For example, to make it more "hash-like" for testing distribution:
+        let mut full_seed = [0u8; 32];
+        full_seed[0] = seed;
+        full_seed[1] = seed.wrapping_add(1);
+        full_seed[2] = seed.wrapping_add(2);
+        // ... and so on, or use a seeded RNG to fill `bytes`.
+        // For this test, simple differentiation is likely sufficient.
```

Comment:
Why not sha256 the seed? Just curious, what's here is fine too.

Reply:
`full_seed` was dead code so I just removed it, the above was sufficient for the simple `u8` differentiator (and makes it easy to see which seed caused any potential issues directly in the block hash). [c8db5e9](https://github.com/stacks-sbtc/sbtc/pull/1727/commits/c8db5e938b17effb2ab1211414b257afd5218064)